lib/scanner/walk.go

```diff
@@ -188,14 +188,13 @@ func (w *walker) walk(ctx context.Context) chan ScanResult {
 			}
 		}()
 
-	loop:
+		// We do not abort on context here because the link between the
+		// filesystem walker and the hasher must not be interrupted, otherwise
+		// the hasher might close finishedChan (returned to caller) before
+		// the filesystem walker is done.
```

Comment:
I don't think that can happen? The walk is done as all files have been added to the `filesToHash` slice in the loop above on line 152-155. It shouldn't matter whether we send all of them or not

Reply:
Right - diff getting smaller again.